clang/lib/AST/CommentParser.cpp

```diff
@@ -75,6 +75,25 @@ class TextTokenRetokenizer {
     return *Pos.BufferPtr;
   }
 
+  char peekNext(unsigned offset) const {
+    assert(!isEnd());
+    assert(Pos.BufferPtr != Pos.BufferEnd);
+    if (Pos.BufferPtr + offset <= Pos.BufferEnd) {
```

Comment:
Shouldn't this be `if (Pos.BufferPtr + offset < Pos.BufferEnd)` (unless this depends of the buffer being null terminated)? Otherwise this allows for an `offset` which results in `Pos.BufferEnd` being read.

Reply:
Good catch, fixed.